index.js

```diff
@@ -22,63 +31,54 @@ program
   .command("add <tasks>")
   .description("Add new task")
   .action((task) => {
-    const todos = readToDos();
-
-    todos.push({ task, done: false });
-    writeToDos(todos);
-    console.log(`Added task: "${task}"`);
+    addTodo(task);
   });
 
 program
   .command("list")
   .description("List all todos")
   .action(() => {
-    const todos = readToDos();
-    if (todos.length === 0) {
-      console.log("No tasks found");
-      return;
-    }
-    todos.forEach((todo, index) =>
-      console.log(`${index + 1}. [${todo.done ? "x" : ""}] ${todo.task}`)
-    );
+    fetchTodos();
   });
 
 program
   .command("done <index>")
   .description("Mark a task as done")
   .action((index) => {
-    const todos = readToDos();
+    const todos = markTaskAsDone(index);
+    if (index > todos.length || index < 1) {
+      console.log("Invalid task number");
+      return;
+    }
+  });
+
+program
+  .command("undone <index>")
+  .description("Mark a task as not done")
+  .action((index) => {
+    const todos = markTaskANotsDone(index);
     if (index > todos.length || index < 1) {
       console.log("Invalid task number");
       return;
     }
-    todos[index - 1].done = true;
-    writeToDos(todos);
-    console.log(`Task ${index} marked as done`);
   });
 
 program
   .command("remove <index>")
   .description("Remove task")
   .action((index) => {
-    const todos = readToDos();
+    const todos = removeTodo(index);
     if (index > todos.length || index < 1) {
       console.log("Invalid task number");
       return;
     }
-    todos.splice(index - 1, 1);
-    writeToDos(todos);
-    console.log(`Task ${index} has been removed`);
   });
```

Comment:
Isn't this check supposed to be before the marking?

Reply:
I added that for when someone mistakenly marks a task as done and want to undo it.